components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/TemplateMgtAPIClient.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.endpoint.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import javax.net.ssl.HttpsURLConnection;
+
+/**
+ * Client for calling /api/identity/template/mgt/v1.0.0/templates/ with mutual ssl authentication
+ */
+public class TemplateMgtAPIClient {
+
+    private static final Log log = LogFactory.getLog(TemplateMgtAPIClient.class);
+
+    private static final String HTTP_METHOD_GET = "GET";
+
+    /**
+     * Send mutual ssl https post request and return data
+     *
+     * @param backendURL URL of the service
+     * @return Received data
+     * @throws IOException
```

Comment:
Exception is not thrown by the method

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/3065.